### DISCUSSION TITLE
How to know if a protocol or plate type exists

### DISCUSSION THREAD

**Discussion by DanRothwell** [ukrobotics/DispenseLibDotNet#20] [Q&A] [answered]
If you try to run a dispense with an invalid protocol ID or plate type ID you get some unhelpful error messages. Either a null reference exception or an error 401.

Is there a way to check that a protocol and plate type exist before trying to run them?

**Comment by mcounsell** [comment, marked as answer]
Please see   _D2DataAccess.GetProtocol( string guid )_   

We don't have a method  _bool ProtocolExists(string guid)_  at this time but if you call the method above you will either get a valid protocol instance OR null.
